instrumentation/java-streams/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/java/inputstream/InputStreamUtils.java

```diff
@@ -50,12 +53,34 @@ public static void addAttribute(Span span, AttributeKey<String> attributeKey, St
     if (span.isRecording()) {
       span.setAttribute(attributeKey, value);
     } else {
-      TRACER
-          .spanBuilder(HypertraceSemanticAttributes.ADDITIONAL_DATA_SPAN_NAME)
-          .setParent(Context.root().with(span))
-          .setAttribute(attributeKey, value)
-          .startSpan()
-          .end();
+      SpanBuilder spanBuilder =
+          TRACER
+              .spanBuilder(HypertraceSemanticAttributes.ADDITIONAL_DATA_SPAN_NAME)
+              .setParent(Context.root().with(span))
+              .setAttribute(attributeKey, value);
+
+      // Also add content type if present
+      if (span.getClass().getName().equals("io.opentelemetry.sdk.trace.SdkSpan")) {
+        try {
+          Method getAttribute =
```

Comment:
Needed to use reflection because this part of code will be running in the application class loader. And that will not have the sdk libraries which has this `SdkSpan` class.

Reply:
Can we store content type in context along with parent span? Rather than retrieving with reflection from span.

Reply:
We can but we will need that change in all the framework instrumentations that we want to support. This was a one place change for this for all parent frameworks

Reply:
What method is this instrumentation on? 

Also how many framework instrumentation points do we support streaming bodies for?

Reply:
> What method is this instrumentation on?

This is instrumented on the `read` method in `java.io.InputStream` 

> Also how many framework instrumentation points do we support streaming bodies for?

We have 4 client instrumentations and all support streaming.

Reply:
Ok, my concern is with overhead, will let you take call on which approach to go. If we go with reflection, should we at least cache the Method object because that wouldn't change, will it? If that's not an expensive operation then nvm.

Reply:
Yes, added the method to static block during instantiation. Will be called only once.